app/models/utility_hookup.rb

```diff
@@ -4,6 +4,7 @@
 class UtilityHookup < ApplicationRecord
   # @return [Space]
   belongs_to :space
+  self.location_parent = :space
```

Comment:
I see that`location_parent` was created for rails routing... but why do we need it?

https://github.com/zinc-collective/convene/blob/53130397f50b27c73cdcfacc0ce53133dc20db3b/app/models/within_location.rb#L6

Reply:
It allows us to provide a url hierarchy when leveraging polymorphic routes. For example: `/spaces/:space_id/utility_hookups` vs `/utility_hookups/`

Reply:
ok within_location.rb  seems like a custom implementation and not something that come with rails.

I'll learn more about polymorphic routes on my own.

Reply:
Yea, rails does not include it by default because most `Rails` applications are flat; and it's [often an anti-pattern to have too deeply nested of resources](https://www.theodinproject.com/lessons/ruby-on-rails-advanced-topics#nested-routes)!

However, we need to keep models within the `Room`  or `Space`; so we almost always have at least those two levels of nesting; plus any additional domain-specific nesting that `Furniture` may want.